### PR TITLE
Update new file to create a new file

### PR DESCRIPTION
**Pull request by lramos15**
Ideally with `New File...` we want the user to also be able to input a filename to create a file.

This will allow the creation of a file with a given file name using a best guess URI.

@bpasero I assigned you since you mentioned the hot exit concerns I figured you might have insight into concerns surrounding the way I select a URI and create a file on disk.

### CODE REVIEW

File: src/vs/workbench/contrib/files/browser/fileCommands.ts [hunk 1]
```diff
@@ -639,19 +641,72 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
...
 	},
-	handler: async (accessor, args?: { languageId?: string; viewType?: string; path?: string }) => {
+	handler: async (accessor, args?: { languageId?: string; viewType?: string; fileName?: string }) => {
 		const editorService = accessor.get(IEditorService);
+		const dialogService = accessor.get(IFileDialogService);
+		const pathService = accessor.get(IPathService);
+		const fileService = accessor.get(IFileService);
+
+		const createFileLocalized = nls.localize('newFileCommand.saveLabel', "Create File");
+		const defaultFileUri = joinPath(await dialogService.defaultFilePath(pathService.defaultUriScheme), args?.fileName ?? 'newfile.txt');
```
Comment (bpasero) [inline src/vs/workbench/contrib/files/browser/fileCommands.ts:698]: Some feedback:
* why the explicit `pathService.defaultUriScheme`?
* should the default file name be `Untitled.txt` to align with untitled files?

File: src/vs/workbench/contrib/files/browser/fileCommands.ts [hunk 2]
```diff
@@ -639,19 +641,72 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
...
+
+		const saveUri = await dialogService.showSaveDialog({ saveLabel: createFileLocalized, title: createFileLocalized, defaultUri: defaultFileUri });
+
+		if (!saveUri) {
+			return;
+		}
+
+		await fileService.createFile(saveUri);
```
Comment (bpasero) [inline src/vs/workbench/contrib/files/browser/fileCommands.ts:706]: Unless you set `overwrite: true`, this will throw an error if the file exists.

**Review by bpasero — CHANGES_REQUESTED** [review]
I like the direction. 

There is now 2 somewhat similar commands for creating a file btw:

image

File: src/vs/workbench/contrib/files/browser/fileCommands.ts (hunk 1 above)
Comment (lramos15) [reply src/vs/workbench/contrib/files/browser/fileCommands.ts:698]: - It expected a scheme and I didn't know if it was safe to pass in file so I just passed in whatever the path service said the default was. Saw this in other places so thought it was the right thing to do.
- `Untitled.txt` makes sense

**Comment by lramos15** [comment]
> I like the direction.
> 
> There is now 2 somewhat similar commands for creating a file btw:
> 
> image

I should look into how to avoid showing this in the command palette since this is only part of the `Create New File....` flow, we don't need yet another new file command showing there

File: src/vs/workbench/contrib/files/browser/fileCommands.ts (hunk 1 above)
Comment (bpasero) [reply src/vs/workbench/contrib/files/browser/fileCommands.ts:698]: I think when you do not specify a default scheme, the file dialog service will just take it, @alexr00 ?

Comment (lramos15) [reply src/vs/workbench/contrib/files/browser/fileCommands.ts:698]: Removed schema filter, seems fine

**Review by bpasero — APPROVED** [review]

**Pull request merged**